dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -1490,4 +1341,176 @@ Set getContainers() {
 	Set getFolders() {
 		return folders;
 	}
+
+	private class DependencyProcessorItem {
+		String assetKey;
+		AssetTypes assetTypes;
+
+		public DependencyProcessorItem(String assetKey, AssetTypes assetTypes) {
+			this.assetKey = assetKey;
+			this.assetTypes = assetTypes;
+		}
+	}
+
+	private class DependencyProcessor {
+		private BlockingQueue<DependencyProcessorItem> queue;
+		private Map<AssetTypes, Set<String>> assetsAlreadyProcessed;
+		private boolean finished;
+		private boolean started;
+		private List<DependencyThread> threads;
+
+		void put(final String assetKey, final AssetTypes assetType) {
+			if (!started) {
+				return;
+			}
+
+			if (assetsAlreadyProcessed != null && !this.alreadyProcess(assetKey, assetType)) {
+				Logger.debug(DependencyProcessor.class, () -> String.format("%s: Putting %s in %s",
+						Thread.currentThread().getName(), assetKey, assetType));
+
+				queue.add(new DependencyProcessorItem(assetKey, assetType));
+				addSet(assetKey, assetType);
+			}
+		}
+
+		private synchronized void addSet(final String assetKey, final AssetTypes assetTypes) {
+			Set<String> set = assetsAlreadyProcessed.get(assetTypes);
+
+			if (set == null) {
+				set = new HashSet<>();
+				assetsAlreadyProcessed.put(assetTypes, set);
+			}
+
+			set.add(assetKey);
+		}
+
+		void start(){
+			Logger.debug(DependencyProcessor.class, "starting");
+			queue = new LinkedBlockingDeque();
+			assetsAlreadyProcessed = new HashMap<>();
+
+			int poolSize = Config.getIntProperty("NUMBER_THREAD_TO_EXECUTE_BUNDLER", 10);
+
+			threads = new ArrayList<>();
+
+			for (int i = 0; i < poolSize; i++) {
```

Comment:
Not sure what are you doing here

Reply:
done https://github.com/dotCMS/core/pull/19955/commits/803a1c5b4aae8f0a873736babf79c2ba79b734b3